src/array_api_stubs/_draft/searching_functions.py

```diff
@@ -139,21 +139,33 @@ def searchsorted(
     """
 
 
-def where(condition: array, x1: array, x2: array, /) -> array:
+def where(
+    condition: array,
+    x1: Union[array, int, float, bool],
+    x2: Union[array, int, float, bool],
+    /,
+) -> array:
     """
     Returns elements chosen from ``x1`` or ``x2`` depending on ``condition``.
 
     Parameters
     ----------
     condition: array
         when ``True``, yield ``x1_i``; otherwise, yield ``x2_i``. Must be compatible with ``x1`` and ``x2`` (see :ref:`broadcasting`).
-    x1: array
-        first input array. Must be compatible with ``condition`` and ``x2`` (see :ref:`broadcasting`).
-    x2: array
-        second input array. Must be compatible with ``condition`` and ``x1`` (see :ref:`broadcasting`).
+    x1: Union[array, int, float, complex, bool]
+        first input array or scalar. Scalar values are treated like an array filled with this value. Must be compatible with ``condition`` and ``x2`` (see :ref:`broadcasting`).
```

Comment:
```suggestion
        first input array. Must be compatible with ``condition`` and ``x2`` (see :ref:`broadcasting`).
```

Aligning with https://github.com/data-apis/array-api/pull/862.